include/gc_inline.h

```diff
@@ -54,6 +58,8 @@
 #ifndef GC_PREFETCH_FOR_WRITE
 # if GC_GNUC_PREREQ(3, 0) && !defined(GC_NO_PREFETCH_FOR_WRITE)
 #   define GC_PREFETCH_FOR_WRITE(x) __builtin_prefetch((x), 1)
+# elif defined(_MSC_VER) && !defined(GC_NO_PREFETCH_FOR_WRITE)
+#   define GC_PREFETCH_FOR_WRITE(x) _mm_prefetch((x), _MM_HINT_T0)
```

Comment:
Work to support WIndows ARM64 recently landed by @jem-patel . Assuming the Mono branch for these prefetch changes has that work, just make sure that platform/arch still builds on CI with these changes. I don't think this intrinsic exists on ARM64.

Reply:
Does this file get included in IL2CPP? If so, we need both ARM32 & ARM64 paths as we support those for UWP. If it's only used by Mono, then we need ARM64 path for Windows Standalone.

From what I can tell looking at the headers, these intrinsics are available there:

    __MACHINEARM_ARM64(void __cdecl __prefetch(const void *))
    __MACHINEARM64(void __cdecl __prefetch2(const void *, uint8_t prfop))
    __MACHINEARM(void __cdecl __prefetchw(const void *))
    
I don't know which one(s) we want to use.

Reply:
My proposal is this
- we have shipped without any of this for the longest time,
- I have no idea which of the ARM intrinsics to use and I have no machine available for testing as of writing,
- hence we should exclude ARM, i.e. wrap in `!defined(_M_ARM) && !defined(_M_ARM64)`

Thoughts?